modules/tinymce/src/plugins/imagetools/main/ts/ui/Buttons.ts

```diff
@@ -5,62 +5,85 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Fun } from '@ephox/katamari';
+
 import Editor from 'tinymce/core/api/Editor';
 import { Toolbar } from 'tinymce/core/api/ui/Ui';
 
 import * as Actions from '../core/Actions';
 
 const register = (editor: Editor) => {
+
+  const buttonApiMap = new Map<string, Toolbar.ToolbarButtonInstanceApi>();
+
   const cmd = (command: string) => () => editor.execCommand(command);
 
-  const onSetup = (buttonApi: Toolbar.ToolbarButtonInstanceApi) => {
-    const setDisabled = () => {
-      const disabled = Actions.getSelectedImage(editor).forall((element) => {
-        return Actions.getEditableImage(editor, element.dom).isNone();
-      });
-      buttonApi.setDisabled(disabled);
-    };
+  const setDisabled = () => {
+    const disabled = Actions.getSelectedImage(editor).forall((element) => {
+      return Actions.getEditableImage(editor, element.dom).isNone();
+    });
 
-    editor.on('NodeChange', setDisabled);
+    buttonApiMap.forEach((api: Toolbar.ToolbarButtonInstanceApi) => {
+      api.setDisabled(disabled);
+    });
+  };
 
-    return () => {
-      editor.off('NodeChange', setDisabled);
-    };
+  const addToButtonApiMap = (key: string, api: Toolbar.ToolbarButtonInstanceApi) => {
+    if (!buttonApiMap.has(key)) {
+      buttonApiMap.set(key, api);
+    }
   };
 
+  editor.on('NodeChange', setDisabled);
```

Comment:
Should we have an `editor.off()` somewhere?! If I add it here it doesn't trigger the `NodeChange` at all.

Reply:
It should be fine to not include that, as since it's an editor managed event it will be cleaned up when the editor is removed.